src/main/java/it/smartcommunitylab/aac/webauthn/persistence/WebAuthnYubicoCredentialsRepository.java

```diff
@@ -0,0 +1,103 @@
+package it.smartcommunitylab.aac.webauthn.persistence;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import com.yubico.webauthn.CredentialRepository;
+import com.yubico.webauthn.RegisteredCredential;
+import com.yubico.webauthn.data.ByteArray;
+import com.yubico.webauthn.data.PublicKeyCredentialDescriptor;
+import com.yubico.webauthn.data.PublicKeyCredentialType;
+
+/**
+ * For this class, everytime we should use the 'username', we
+ * will instead use a string in the following format:
+ * 'realmname{@link #separator}username'. This
+ * is due to the fact that yubico's library assumes that two distinct users can
+ * not have the same username (https://git.io/JD5Vr).
+ */
+public class WebAuthnYubicoCredentialsRepository implements CredentialRepository {
+
+    private final String providerId;
+    WebAuthnUserAccountRepository userAccountRepository;
```

Comment:
visibility private for all props

Reply:
Fixed in 564dd775b4921c700d82586e3fe98ed38e3ad80d